File: 3-int-to-bool/solutions/go/problem1.go
package main

import "fmt"

func isPositive(num int) bool {
	return false
}

func main() {
	// test function isPositive
	// int -> bool

	tests := []test{
		{-4, false},
		{0, false},
		{10, true},
	}

	runTests(tests)
}

type test struct {
	input    int
	expected bool
}

func runTests(tests []test) {
	fmt.Println()
	for _, test := range tests {
		result := isPositive(test.input)

		output := testString(test.input, result)
		output += okString(result, test.expected)
		fmt.Println(output)
	}
}

func testString(input int, result bool) string {
	str := "isPositive(%d) \t -> %t "
	return fmt.Sprintf(str, input, result)
}

func okString(result bool, expected bool) string {
	var okStr string
	if result == expected {
		okStr = "\t (OK)"
	} else {
		okStr = fmt.Sprintf("\t (X) expected %t \n", expected)
	}

	return okStr
}

File: 3-int-to-bool/solutions/java/Problem1.java
class Problem1 {
    public static boolean isPositive(int num) {
        return num > 0;
    }

    public static void main(String[] args) {

        Test[] tests = {
            new Test(-4, false), 
            new Test(0, false), 
            new Test(10, true)
            };
        
        runTests(tests);
    }

    public static void runTests(Test[] tests){
        System.out.println();
        for (Test test : tests) {
            boolean result = isPositive(test.input);
            String output = String.format("isPositive(%d) \t ->  %b ", test.input, result);

            if (result == test.expected){
                output += "\t (OK)";
            } else {
                output += String.format("\t(X) expected %b %n", test.expected);
            }

            System.out.println(output);
        }
    }

        static class Test {
            int input;
            boolean expected;

            Test(int input, boolean expected) {
                this.input = input;
                this.expected = expected;
            }
        }

}